pkg/scheduler/framework/plugins/nodevolumelimits/csi.go

```diff
@@ -162,7 +167,44 @@ func (pl *CSILimits) filterAttachableVolumes(
 			pvcName = ephemeral.VolumeClaimName(pod, &vol)
 			isEphemeral = true
 		default:
+			// Inline Volume does not have PVC.
+			// Need to check if CSI migration is enabled for inline volume.
+			isMigratable = pl.translator.IsInlineMigratable(&vol)
+			// If the volume is not migratable, it will be count in non_csi filter.
+			if !isMigratable {
+				continue
+			}
+		}
+
+		if isMigratable {
```

Comment:
isMigratable is default to false so it will always continue even if it is in these two switch conditions
```
switch {
		case vol.PersistentVolumeClaim != nil:
		case vol.Ephemeral != nil:
```

So we have to make explicitly check on the default condition to skip the rest checking.

Reply:
Hi @alculquicondor I think this is ready for review again.

Reply:
It might be easier to understand if you look at the test cases. But the tldr is that inline volumes are not being counted when we try to find attachablevolumes after csi migration is enabled.

So our logic here is if it is not PVC or Ephemeral volume, then it is an intree volume. We need to make sure if this inline volume is a migratble plugin and migration is on for it. because all other types have a PVC but the inline one. That's why we need this special check in the default section.

The volume in the other switch section like PVC and Ephemeral are by nature not part of CSI migration target. So we will just skip them.

https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/625-csi-migration/README.md This has more context on csi migration if you are interested.

Reply:
Thanks for the explanation and for bearing with me.

A nit to improve readability: Put all the block inside `if isMigratable` into a function and call it inside the switch function.
Then it will be much more clear that you don't need to do anything else for inline volumes.

Reply:
It is indeed looking more readable. Thanks